File: src/python/PythonSDK/foundationallm/models/authentication/authentication_types.py
from enum import Enum

class AuthenticationTypes(str, Enum):
    """Enumerator of the Authentication providers."""
    UNKNOWN = "Unknown"
    AZURE_IDENTITY = "AzureIdentity"
    API_KEY = "APIKey"
    CONNECTION_STRING = "ConnectionString"
    ACCOUNT_KEY = "AccountKey"

File: src/dotnet/Common/Constants/Authentication/AuthenticationTypes.cs
namespace FoundationaLLM.Common.Constants.Authentication
{
    /// <summary>
    /// Authentication types for API Endpoints
    /// </summary>
    public enum AuthenticationTypes
    {
        /// <summary>
        /// Unknown authentication type.
        /// </summary>
        Unknown = -1,

        /// <summary>
        /// Azure managed identity authentication type.
        /// </summary>
        AzureIdentity,

        /// <summary>
        /// API key authentication type.
        /// </summary>
        APIKey,

        /// <summary>
        /// Connection string authentication type.
        /// </summary>
        ConnectionString,

        /// <summary>
        /// Account key authentication type.
        /// </summary>
        AccountKey
    }
}
